From 9061ee45b8dbe5431c23e3f628089d703ccad0b1 Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston <jeremyhu@apple.com>
Date: Thu, 8 Mar 2012 00:50:13 -0800
Subject: [PATCH 7/8] darwin: Use read(2) rather than recv(2)

2dcf8b025be88a25d4333abdc28d425b88238d96 was causing some regressions on
darwin, so go back to using read(2) there until I have time to investigate
further.

Signed-off-by: Jeremy Huddleston <jeremyhu@apple.com>
---
 src/xcb_in.c |    8 ++++++++
 1 file changed, 8 insertions(+)

diff --git libxcb-1.8/src/xcb_in.c libxcb-1.8/src/xcb_in.c
index 969cfc0..fdcc813 100644
--- libxcb-1.8/src/xcb_in.c
+++ libxcb-1.8/src/xcb_in.c
@@ -269,7 +269,11 @@ static int read_block(const int fd, void *buf, const ssize_t len)
     int done = 0;
     while(done < len)
     {
+#ifdef __APPLE__
+        int ret = read(fd, ((char *) buf) + done, len - done);
+#else
         int ret = recv(fd, ((char *) buf) + done, len - done,MSG_WAITALL);
+#endif
         if(ret > 0)
             done += ret;
 #ifndef _WIN32
@@ -661,7 +665,11 @@ void _xcb_in_replies_done(xcb_connection_t *c)
 
 int _xcb_in_read(xcb_connection_t *c)
 {
+#ifdef __APPLE__
+    int n = read(c->fd, c->in.queue + c->in.queue_len, sizeof(c->in.queue) - c->in.queue_len);
+#else
     int n = recv(c->fd, c->in.queue + c->in.queue_len, sizeof(c->in.queue) - c->in.queue_len,MSG_WAITALL);
+#endif
     if(n > 0)
         c->in.queue_len += n;
     while(read_packet(c))
-- 
1.7.9.2

